CLUE_Rock_Paper_Scissors/advanced/clue-multi-rpsgame.py

```diff
@@ -0,0 +1,556 @@
+    _button_a = digitalio.DigitalInOut(board.BUTTON_A)
+    _button_a.switch_to_input(pull=digitalio.Pull.UP)
+    _button_b = digitalio.DigitalInOut(board.BUTTON_B)
+    _button_b.switch_to_input(pull=digitalio.Pull.UP)
+    button_left = lambda: not _button_a.value
+    button_right = lambda: not _button_b.value
+    ##button_left = lambda: clue.button_a
+    ##button_right = lambda: clue.button_b
+
+blankScreen(display, pixels)
+
+# Set to True for blue flashing when devices are joining the playing group
+JG_FLASH = False
+
+IMAGE_DIR = "rps/images"
+AUDIO_DIR = "rps/audio"
+
+files = (("searching", "welcome-to", "arena", "ready")
+         + ("rock", "paper", "scissors")
+         + ("start-tx", "end-tx", "txing")
+         + ("rock-scissors", "paper-rock", "scissors-paper")
+         + ("you-win", "draw", "you-lose", "error")
+         + ("humiliation", "excellent"))
+
+gc.collect()
+d_print(2, "GC before SJ", gc.mem_free())
+sample = SampleJukebox(audio_out, files,
+                       directory=AUDIO_DIR, error_output=True)
+del files  # not needed anymore
+gc.collect()
+d_print(2, "GC after SJ", gc.mem_free())
+
+# A lookup table in Dict form for win/lose
+# does not need to cater for draw condition
+WAV_VICTORY_NAME = { "rp": "paper-rock",
+                     "pr": "paper-rock",
+                     "ps": "scissors-paper",
+                     "sp": "scissors-paper",
+                     "sr": "rock-scissors",
+                     "rs": "rock-scissors"}
+
+# This limit is based on displaying names on screen with scale=2 font
+MAX_PLAYERS = 8
+# Some code is dependent on these being lower-case
+CHOICES = ("rock", "paper", "scissors")
+
+rps_display = RPSDisplay(display, pixels,
+                         CHOICES, sample, WAV_VICTORY_NAME,
+                         MAX_PLAYERS, BUTTON_Y_POS,
+                         IMAGE_DIR + "/rps-sprites-ind4.bmp",
+                         ble_color=JG_RX_COL)
+
+# Transmit maximum times in seconds
+JG_MSG_TIME_S = 20
+FIRST_MSG_TIME_S = 12
+STD_MSG_TIME_S = 4
+LAST_ACK_TIME_S = 1.5
+
+
+# Intro screen with audio
+rps_display.introductionScreen()
+
+# Enable the Bluetooth LE radio and set player's name (from secrets.py)
+ble = BLERadio()
+if ble_name is not None:
+    ble.name = ble_name
+
+
+game_no = 1
+round_no = 1
+wins = 0
+losses = 0
+draws = 0
+voids = 0
+
+# TOTAL_ROUNDS = 5
+TOTAL_ROUNDS = 3
+
+CRYPTO_ALGO = "chacha20"
+KEY_SIZE = 8  # in bytes
+KEY_ENLARGE = 256 // KEY_SIZE // 8
+
+
+def evaluateRound(mine, yours):
+    """Determine who won the game based on the two strings mine and yours.
+       Returns three booleans (win, draw, void)."""
+    # Return with void at True if any input is None
+    try:
+        mine_lc = mine.lower()
+        yours_lc = yours.lower()
+    except AttributeError:
+        return (False, False, True)
+
+    r_win = r_draw = r_void = False
+    # pylint: disable=too-many-boolean-expressions
+    if (mine_lc == "rock" and yours_lc == "rock"
```

Comment:
Could use `in` here to check that the values are valid. Then just check `mine_lc == yours_lc`  for the draw case.

Reply:
Done